test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java

```diff
@@ -22,45 +22,93 @@
 
-@Test(groups = { "level.extended" })
+/**
+ * This test StackWalker features added in Java 10 and is supplementary to the Java 9 tests.
+ *
+ */
 @SuppressWarnings("nls")
 public class StackWalkerTestJava10 {
 	static final String TEST_MYLOADER_INSTANCE = "TestMyloaderInstance";
 	protected static Logger logger = Logger.getLogger(StackWalkerTestJava10.class);
 
-	@Test
+	@Test(groups = { "level.extended" })
 	public void testSanity() {
-		f1(3);
+		sanityMethod_1(3);
 	}
 
-	void f1(int i) {
-		if (i > 0) f1(i-1); else f2(new Integer(42));
+	void sanityMethod_1(int i) {
+		if (i > 0) sanityMethod_1(i-1); else sanityMethod_2(new Integer(42));
 	}
 
-	private void f2(Integer in) {
-		f3(in, new HashMap<String, Integer>());
+	private void sanityMethod_2(Integer in) {
+		sanityMethod_3(in, new HashMap<String, Integer>());
 	}
 
-	private int f3(Integer in, HashMap<String, Integer> hashMap) {
+	private int sanityMethod_3(Integer in, HashMap<String, Integer> hashMap) {
 		printStack("Hello");
 		return 42;
 	}
 
 	public void printStack(String arg) {
-		List<StackFrame> frameList = (StackWalker.getInstance(Option.RETAIN_CLASS_REFERENCE))
-		.walk(s -> s.collect(Collectors.toList()));
-		String expectedDescriptors[];
-		for (StackFrame f: frameList) {
-			System.out.println(f.getDescriptor());
+		final ClassLoader myClassLoader = this.getClass().getClassLoader();
+		List<StackFrame> frameList = 
+				StackWalker.getInstance(Option.RETAIN_CLASS_REFERENCE)
+				.walk(s -> s.collect(Collectors.toList()));
+		String expectedDescriptors[] = {
+				"(Ljava/lang/String;)V",
+				"(Ljava/lang/Integer;Ljava/util/HashMap;)I",
+				"(Ljava/lang/Integer;)V",
+				"(I)V",
+				"(I)V",
+				"(I)V",
+				"(I)V",
+				"()V"
+		};
+		Iterator<StackFrame> frameIterator = frameList.iterator();
+		for (String s: expectedDescriptors) {
+			assertTrue(frameIterator.hasNext(), "too few frames");
+			StackFrame f = frameIterator.next();
+			final String actualDescriptor = f.getDescriptor();
+			assertEquals(actualDescriptor, s, "wrong descriptor");
+			assertEquals(MethodType.fromMethodDescriptorString(s,myClassLoader), f.getMethodType(), 
+					"wrong descriptor");			
 		}
 	}
 
-}
+	@Test(groups = { "level.extended" })
+	public void testRetainOption() {
+		StackWalker walker = StackWalker.getInstance();
+		getMyMethodType(walker);
+	}
+
+	String getMyMethodType(StackWalker walker) {
+		Optional<StackFrame> callerFrameOption = walker.walk(s->s.findFirst());
+		assertTrue(callerFrameOption.isPresent(), "missing stack frame");
+		StackFrame callerFrame = callerFrameOption.get();
+		assertEquals(callerFrame.getDescriptor(), "(Ljava/lang/StackWalker;)Ljava/lang/String;", 
+				"wrong descriptor when RETAIN_CLASS_REFERENCE not configured");
+		boolean thrown = false;
+		try {
+			callerFrame.getMethodType();
+		} catch (UnsupportedOperationException e) {
+			thrown = true;
+		}
+		assertTrue(thrown, "UnsupportedOperationException not thrown");
```

Comment:
the pattern where a test is expected to throw an exception can be written more cleanly as:
```
try {
   some code..
  fail("Expected UnsupportedOperationException");
} catch(UnsupportedOperationException e) {
  /* Success */
}
```

Reply:
Good plan.  Done.